src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -224,6 +225,18 @@ def get_location_type_with_default_from_resource_group(cli_ctx):
     options_list=['--object-id', '-i'],
     help='The unique ID of the Azure AD administrator.')
 
+read_scale_param_type = CLIArgumentType(
+    options_list=['--read-scale'],
+    help='If enabled, connections that have application intent set to readonly '
+    'in their connection string may be routed to a readonly secondary replica. '
+    'This property is only settable for Premium and Business Critical databases.',
+    arg_type=get_enum_type(DatabaseReadScale))
+
+read_replicas_param_type = CLIArgumentType(
+    options_list=['--read-replicas'],
+    help='The number of readonly replicas to provision for the database. '
```

Comment:
According to my understanding, if I don't set this parameter but enable read scale, readReplicaCount will be set to 1 and if I disable read scale, readReplicaCount will be set to 0, right? If yes, Can you show up such behavior for this parameter and default value?

Reply:
The parameters are independent and depend on the database edition.  Only one of the parameters matters for each edition.  The other is ignored.  Please see my other reply on prior comment and also the help text for the parameters:

read_scale_param_type = CLIArgumentType(
    options_list=['--read-scale'],
    help='If enabled, connections that have application intent set to readonly '
    'in their connection string may be routed to a readonly secondary replica. '
    'This property is only settable for Premium and Business Critical databases.',
    arg_type=get_enum_type(DatabaseReadScale))

read_replicas_param_type = CLIArgumentType(
    options_list=['--read-replicas'],
    help='The number of readonly replicas to provision for the database. '
    'Only settable for Hyperscale edition.')